lib/dep_selector/selector.rb

```diff
@@ -67,11 +67,9 @@ def find_solution(solution_constraints, valid_packages = nil)
       packages_to_include_in_solve = trim_unreachable_packages(dep_graph, solution_constraints)
 
       begin
-        Timeout::timeout(@time_bound, Exceptions::TimeBoundExceeded) do
-          # first, try to solve the whole set of constraints
-          solve(dep_graph.clone, solution_constraints, valid_packages, packages_to_include_in_solve)
-        end
-      rescue Exceptions::NoSolutionFound
+        # first, try to solve the whole set of constraints
+        solve(dep_graph.clone, solution_constraints, valid_packages, packages_to_include_in_solve)
+      rescue Exceptions::NoSolutionFound, Exceptions::TimeBoundExceededNoSolution
         # since we're here, solving the whole system failed, so add
         # the solution_constraints one-by-one and try to solve in
         # order to find the constraint that breaks the system in order
```

Comment:
The failures I've seen seem to be a function of constraint structure, not size, specifically where a runlist cookbook is is forcing a common cookbook to an older version, while other runlist cookbooks want that cookbook at latest. Sometimes those have a solution, but are slow, while others are simply insoluble. The difference isn't that huge; the same basic root cause drives both.

While the constraint cutting logic here is imperfect, it's currently the best option for